dev/release_notes.readme.md

```diff
@@ -0,0 +1,102 @@
+
+| | subheading | labels | meaning |
+| - |------------- | ------- | -------- |
+| 1 | New features and major changes | `release notes: highlight` | PRs introducing changes that should be highlighted at the top of the release notes |
+| 2 | Fixed bugs that could lead to incorrect results | `kind: bug: wrong result` | PRs fixing bugs that result in mathematically or otherwise wrong results |
+| 3 | Removed or obsolete functionality | `topic: obsolete functionality` | PRs changing `lib/obsolete.g*` (move functions there or remove something from there) |
+| 4 | Fixed bugs that could lead to crashes or unexpected error messages | `kind: bug: crash` | PRs fixing bugs that may cause GAP to crash |
+| 4 | Fixed bugs that could lead to crashes or unexpected error messages | `kind: bug: unexpected error` | PRs fixing bugs that may cause GAP to run into an unexpected error |
+| 5 | Other fixed bugs | `kind: bug:` | PRs fixing bugs |
+| 6 | Improved and extended functionality | `kind: enhancement` OR `kind: new feature` OR `kind: performance` | PRs implementing enhancements |
+| 7 | Improvements in the experimental way to allow 3rd party code to link GAP as a library (libgap) | `topic: libgap` | PRs that are related to libgap |
+| 8 | Improvements in for the **Julia** integration | `topic: julia` | PRs that are related to the Julia integration |
+| 9 | Changed documentation | `topic: documentation` | PRs improving the documentation |
+| 10 | Packages | `topic: packages` | PRs related to package handling, or specific to a package (for packages w/o issue tracker) |
+| 11 | Other changes | `release notes: needed` OR `release notes: added` | PRs relevant for the release notes but without a label that fits --in each case, think about assigning such a label or introducing a new one |
+
+- For empty groups of pull requests, the subheading need not appear in `CHANGES.md`.
+
+- If the group "Other changes" is too large then we should decide whether other labels (most likely some `topic: ...` labels) can be used to define useful groups; in this case, a subheading has to be defined for this label.
+  (There will be a script for creating the current `CHANGES.md` section in the `dev` subdirectory of the main GAP directory, the change of the rules should be made in this script.)
+
+- Concerning the ordering, I think that somebody who reads the release notes of a new GAP version will be first interested in highlights, then in bugs which may affect one's computations (wrong results), then in removed functionality (which may break old private code), then in general improvements.
+
+  I think there is no way to satisfy all perspectives of what is most important, in particular concerning the `topic: ...` labels.
+  Note that `CHANGES.md` is just **one** way to present the changes, eventually we should put the source data (pull request number, release notes text, labels) into the GAP distribution, and provide a tool for evaluating the data inside a GAP session.
```

Comment:
If we start a PR for this, we should add a link here.

Reply:
I think that as soon as we have the files with the data list(s) and the GAP function (most likely in the `dev` directory of the main GAP directory), we should mention these files.